metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -146,48 +151,72 @@ def build_plan(
             logger.info(f"Generating compute metrics node for {metric_spec}")
-            metric = self._metric_semantics.get_metric(metric_spec)
-            metric_input_measure_specs = self._metric_semantics.measures_for_metric(metric_spec)
-
-            logger.info(
-                f"For {metric_spec}, needed measures are:\n"
-                f"{pformat_big_objects(metric_input_measure_specs=metric_input_measure_specs)}"
-            )
+            blank_metric_spec = MetricSpec.from_element_name(metric_spec.element_name)
+            metric = self._metric_semantics.get_metric(blank_metric_spec)
+            if metric.type == MetricType.DERIVED:
+                metric_input_specs = self._metric_semantics.metric_input_specs_for_metric(blank_metric_spec)
+                logger.info(
+                    f"For derived metric: {metric_spec}, needed metrics are:\n"
+                    f"{pformat_big_objects(metric_input_specs=metric_input_specs)}"
+                )
 
-            combined_where = query_spec.where_constraint
-            if metric.constraint:
-                metric_constraint = WhereConstraintConverter.convert_to_spec_where_constraint(
-                    self._data_source_semantics, metric.constraint
+                compute_metric_node = self._build_metrics_output_node(
+                    metric_specs=metric_input_specs,
+                    queried_linkable_specs=queried_linkable_specs,
+                    where_constraint=where_constraint,
+                    time_range_constraint=time_range_constraint,
                 )
-                if combined_where:
-                    combined_where = combined_where.combine(metric_constraint)
-                else:
-                    combined_where = metric_constraint
-
-            aggregated_measures_node = self.build_aggregated_measures(
-                metric_input_measure_specs=metric_input_measure_specs,
-                queried_linkable_specs=query_spec.linkable_specs,
-                where_constraint=combined_where,
-                time_range_constraint=query_spec.time_range_constraint,
-                cumulative=metric.type == MetricType.CUMULATIVE,
-                cumulative_window=metric.type_params.window if metric.type == MetricType.CUMULATIVE else None,
-                cumulative_grain_to_date=(
-                    metric.type_params.grain_to_date if metric.type == MetricType.CUMULATIVE else None
-                ),
-            )
-            compute_metrics_nodes.append(
-                self.build_computed_metrics_node(
-                    metric_spec=metric_spec,
-                    aggregated_measures_node=aggregated_measures_node,
+                compute_metrics_nodes.append(
+                    ComputeMetricsNode[SqlDataSetT](
+                        parent_node=compute_metric_node,
+                        metric_specs=[metric_spec],
+                    )
                 )
-            )
+            else:
```

Comment:
same logic as the prior build compute metric node but placed into this `else` block